streams/src/main/java/org/apache/kafka/streams/state/internals/LogicalKeyValueSegment.java

```diff
@@ -176,7 +184,7 @@ public synchronized KeyValueIterator<Bytes, byte[]> range(final Bytes from, fina
         // with empty bytes from the returned iterator. this filtering is accomplished by
         // passing the prefix filter into StrippedPrefixKeyValueIteratorAdapter().
         final Bytes toBound = to == null
-            ? Bytes.increment(prefixKeyFormatter.getPrefix())
+            ? incrementWithoutOverflow(prefixKeyFormatter.getPrefix())
```

Comment:
> a range query with `toBound == null` means

You mean `to` here, which is the input parameter (not `toBound` which is the physical bound for the store)?

> Except in the edge case where incrementing the prefix would cause overflow, in which case using null for the upper bound is correct. 

But if we overflow, and set `toBound = null` (as returned by `incrementWithoutOverflow()`) would it not imply we scan the physical store to its end, beyond the current logical segment bound?

Reply:
Ah. Thanks for clarifying -- this make sense!